### PR TITLE
Search-356

### PR DESCRIPTION
**Pull request by MBkkt**
### Scope & Purpose

- [ ] :hankey: Bugfix
- [ ] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.10: *(Please link PR)*
  - [ ] Backport for 3.9: *(Please link PR)*
  - [ ] Backport for 3.8: *(Please link PR)*

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket:
- [ ] Design document:

### CODE REVIEW

File: lib/Geo/ShapeContainer.cpp [hunk 1]
```diff
@@ -58,15 +59,11 @@ namespace {
...
-  v.emplace_back(rect->GetVertex(2).ToPoint());
-  v.emplace_back(rect->GetVertex(3).ToPoint());
-  v.emplace_back(rect->GetVertex(0).ToPoint());
-  std::unique_ptr<S2Loop> loop;
-  loop = std::make_unique<S2Loop>(std::move(v), S2Debug::DISABLE);
+  auto first = rect->GetVertex(0).ToPoint();
+  std::array<S2Point, 5> v{first, rect->GetVertex(1).ToPoint(),
+                           rect->GetVertex(2).ToPoint(),
+                           rect->GetVertex(3).ToPoint(), first};
+  auto loop = std::make_unique<S2Loop>(v, S2Debug::DISABLE);
```
Comment (MBkkt) [inline lib/Geo/ShapeContainer.cpp:66]: S2Loop ctor accept span

File: lib/Geo/ShapeContainer.cpp [hunk 2]
```diff
@@ -676,20 +673,24 @@ bool ShapeContainer::intersects(S2Polyline const* other) const {
...
+
 bool intersectRectPolygon(S2LatLngRect const* rect, S2Polygon const* poly) {
   // only used in legacy situations
-  if (rect->is_full()) {
-    return true;  // rectangle spans entire sphere
-  } else if (rect->is_point()) {
-    return poly->Contains(rect->lo().ToPoint());  // easy case
-  } else if (!rect->Intersects(poly->GetRectBound())) {
-    return false;  // cheap rejection
+  if (ADB_UNLIKELY(rect->is_point())) {
```
Comment (Dronplane) [inline lib/Geo/ShapeContainer.cpp:679]: NIT 
Why is this "unlikely" and others are "likely"?  I assume it depends on the data in the collection.

File: arangod/Aql/IResearchViewNode.cpp [hunk 3]
```diff
@@ -2160,4 +2160,13 @@ IResearchViewNode::OptimizationState::replaceAllViewVariables(
   return uniqueVariables;
 }
 
+bool IResearchViewNode::isBuilding() const {
+  if (!_view || !ServerState::instance()->isCoordinator() ||
```
Comment (Dronplane) [inline arangod/Aql/IResearchViewNode.cpp:2164]: "!ServerState::instance()->isCoordinator()"
I guess we should also report "is building" on the single server.

File: arangod/IResearch/IResearchViewCoordinator.h [hunk 4]
```diff
@@ -126,13 +127,24 @@ class IResearchViewCoordinator final : public LogicalView {
...
+    Data(std::string cname, VPackBuilder&& builder, bool building)
+        : collectionName{std::move(cname)},
+          definition{std::move(builder)},
+          isBuilding{building} {}
+
+    std::string collectionName;
+    VPackBuilder definition;
+    bool isBuilding{false};
+  };
+  containers::FlatHashMap<DataSourceId, Data> _collections;
```
Comment (gnusi) [inline arangod/IResearch/IResearchViewCoordinator.h:147]: Probably NodeHashMap would work better given the size of the value type

Comment (MBkkt) [reply arangod/IResearch/IResearchViewCoordinator.h]: Done

File: tests/js/common/aql/aql-view-arangosearch-ddl-cluster.js [hunk 5]
```diff
@@ -69,14 +77,48 @@ function IResearchFeatureDDLTestSuite() {
       }
     },
 
+    testViewIsBuilding: function () {
+      if (isServer) {
```
Comment (Dronplane) [inline tests/js/common/aql/aql-view-arangosearch-ddl-cluster.js:81]: Suggest to add `internal.debugClearFailAt();` to make test stable for failures in other

**Review by Dronplane — APPROVED** [review]
LGTM

**Pull request merged**